src/context/client.rs

```diff
@@ -19,48 +20,92 @@ impl RedisModuleClientInfo {
+/// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
 impl Context {
     pub fn get_client_id(&self) -> u64 {
         unsafe { RedisModule_GetClientId.unwrap()(self.ctx) }
     }
 
-    pub fn get_client_name(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById
+    pub fn get_client_name_by_id(&self, client_id: u64) -> ValkeyString {
         let client_name = unsafe { RedisModule_GetClientNameById.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_name)
     }
 
-    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById using current client ID
+    pub fn get_client_name(&self) -> ValkeyString {
+        self.get_client_name_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_SetClientNameById
+    pub fn set_client_name_by_id(&self, client_id: u64, client_name: &ValkeyString) -> i64 {
         let resp = unsafe { RedisModule_SetClientNameById.unwrap()(client_id, client_name.inner) };
         resp as i64
     }
 
-    pub fn get_client_username(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
-        let client_username =
-            unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
-        ValkeyString::from_redis_module_string(self.ctx, client_username)
+    /// wrapper for RedisModule_SetClientNameById using current client ID
+    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
+        self.set_client_name_by_id(self.get_client_id(), client_name)
     }
 
+    /// wrapper for RedisModule_GetClientUserNameById
     pub fn get_client_username_by_id(&self, client_id: u64) -> ValkeyString {
         let client_username =
             unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_username)
     }
 
+    /// wrapper for RedisModule_GetClientUserNameById using current client ID
+    pub fn get_client_username(&self) -> ValkeyString {
+        self.get_client_username_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_GetClientCertificate
     pub fn get_client_cert(&self) -> ValkeyString {
         let client_id = self.get_client_id();
         let client_cert = unsafe { RedisModule_GetClientCertificate.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_cert)
     }
 
-    pub fn get_client_info(&self) -> RedisModuleClientInfo {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientInfoById
+    pub fn get_client_info_by_id(&self, client_id: u64) -> RedisModuleClientInfo {
```

Comment:
Can we handle the error cases of this API?
```
int VM_GetClientInfoById(void *ci, uint64_t id) {
    client *client = lookupClientByID(id);
    if (client == NULL) return VALKEYMODULE_ERR;
    if (ci == NULL) return VALKEYMODULE_OK;

    /* Fill the info structure if passed. */
    uint64_t structver = ((uint64_t *)ci)[0];
    return modulePopulateClientInfoStructure(ci, client, structver);
}
```

Reply:
updated the code but still thinking about it